File: 25. Exam Preparation/Bunny-Wars - Author's solutions/Java-Skeleton-Eclipse/src/main/BunnyWarsStructure.java
package main;

import java.util.*;

public class BunnyWarsStructure implements IBunnyWarsStructure {

    private static final int TeamCount = 5;

    @Override
    public int getBunnyCount() {
        throw new UnsupportedOperationException();
    }

    @Override
    public int getRoomCount() {
        throw new UnsupportedOperationException();
    }

    @Override
    public void addRoom(int roomId) {
        throw new UnsupportedOperationException();
    }

    @Override
    public void addBunny(String name, int team, int roomId) {
        throw new UnsupportedOperationException();
    }

    @Override
    public void remove(int roomId) {
        throw new UnsupportedOperationException();
    }

    @Override
    public void next(String bunnyName) {
        throw new UnsupportedOperationException();
    }

    @Override
    public void previous(String bunnyName) {
        throw new UnsupportedOperationException();
    }

    @Override
    public void detonate(String bunnyName) {
        throw new UnsupportedOperationException();
    }

    @Override
    public Iterable<Bunny> listBunniesByTeam(int team) {
        throw new UnsupportedOperationException();
    }

    @Override
    public Iterable<Bunny> listBunniesBySuffix(String suffix) {
        throw new UnsupportedOperationException();
    }
}

File: 25. Exam Preparation/Bunny-Wars - Author's solutions/C#-Skeleton/BunnyWars.Core/BunnyWarsStructure.cs
namespace BunnyWars.Core
{
    using System;
    using System.Collections.Generic;

    public class BunnyWarsStructure : IBunnyWarsStructure
    {
        public int BunnyCount { get { throw new NotImplementedException(); } }

        public int RoomCount { get { throw new NotImplementedException(); } }

        public void AddRoom(int roomId)
        {
            throw new NotImplementedException();
        }

        public void AddBunny(string name, int team, int roomId)
        {
            throw new NotImplementedException();
        }

        public void Remove(int roomId)
        {
            throw new NotImplementedException();
        }

        public void Next(string bunnyName)
        {
            throw new NotImplementedException();
        }

        public void Previous(string bunnyName)
        {
            throw new NotImplementedException();
        }

        public void Detonate(string bunnyName)
        {
            throw new NotImplementedException();
        }

        public IEnumerable<Bunny> ListBunniesByTeam(int team)
        {
            throw new NotImplementedException();
        }

        public IEnumerable<Bunny> ListBunniesBySuffix(string suffix)
        {
            throw new NotImplementedException();
        }
    }
}
